fluent-bundle/src/builtins.ts

```diff
@@ -37,12 +54,35 @@ export function NUMBER(
   }
 
   if (arg instanceof FluentNumber || arg instanceof FluentDateTime) {
-    return new FluentNumber(arg.valueOf(), { ...arg.opts, ...values(opts) });
+    return new FluentNumber(arg.valueOf(), {
+      ...arg.opts,
+      ...values(opts, NUMBER_ALLOWED)
+    });
   }
 
   throw new TypeError("Invalid argument to NUMBER");
 }
 
+const DATETIME_ALLOWED = [
+  "dateStyle",
+  "timeStyle",
+  "fractionalSecondDigits",
+  "calendar",
```

Comment:
I'm torn on this one. This sounds like a scary thing for localizers to modify, in particular when they're doing it inconsistently.

I'd go with restricting this for now.